### PR TITLE
Improve: force break after an infix op only if followed by another one

### PR DESCRIPTION
**Pull request by gpetiot**
Forcing the break was a bit too aggressive and generates unnecessary diffs with 0.9
cc @samoht

### CODE REVIEW

File: test/passing/infix_bind.ml [hunk 1]
```diff
@@ -23,36 +23,27 @@ f x
 ;;
 f x >>= fun y ->
 g y >>= fun () ->
-f x >>= fun y ->
-g y >>= fun () ->
-f x >>= fun y ->
-g y >>= fun () ->
-y ()
+f x >>= fun y -> g y >>= fun () -> f x >>= fun y -> g y >>= fun () -> y ()
```
Comment (Julow) [inline test/passing/infix_bind.ml]: This formatting is weird. They should all break.

Comment (gpetiot) [reply test/passing/infix_bind.ml]: I think if the chain is long, there is a good chance one of the bodies will be long enough to break the line.
So the cases we want to improve are indeed the smaller ones.

Comment (gpetiot) [reply test/passing/infix_bind.ml]: So do you think we should not merge this PR?

Comment (Julow) [reply test/passing/infix_bind.ml]: I find this case a bit problematic. What make it hard to fix ?

Comment (gpetiot) [reply test/passing/infix_bind.ml]: It should be better now.

File: src/Fmt_ast.ml [hunk 2]
```diff
@@ -1598,6 +1598,17 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       let xargs, xbody = Sugar.fun_ c.cmts (sub_exp ~ctx r) in
       let indent_wrap = if parens then -2 else 0 in
       let pre_body, body = fmt_body c ?ext xbody in
+      let followed_by_infix_op =
+        match xbody.ast.pexp_desc with
+        | Pexp_apply
+            ( { pexp_desc= Pexp_ident {txt= Lident id; loc= _}
+              ; pexp_attributes= []
+              ; _ }
+            , [(Nolabel, _); (Nolabel, {pexp_desc= Pexp_fun _; _})] )
```
Comment (Julow) [inline src/Fmt_ast.ml]: Missing `Pexp_function`.

Comment (gpetiot) [reply src/Fmt_ast.ml]: Done

**Pull request merged**